### PR TITLE
Finish most of the sockets shims.

### PR DESCRIPTION
**Pull request by pgavlin**
- Shim SocketFlags
- Shim select()
- Fix some issues with error propagation in socket creation and accepted
  sockets

The only functions that have not been shimmed are those relied upon for event notification (epoll/kqueue).

### CODE REVIEW

**Comment by pgavlin** [comment]
@stephentoub @sokket @nguerrera PTAL

**Comment by sokket** [comment]
Looks like Interop.optlevel.cs and Interop.optname.cs don't have any changes but are part of the PR

File: src/Native/System.Native/pal_networking.cpp [hunk 1]
```diff
@@ -6,15 +6,16 @@
...
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <sys/socket.h>
+#include <sys/ioctl.h>
 #include <unistd.h>
-#include <vector>
+#include <alloca.h>
 #include <errno.h>
+#include <assert.h>
+#include <vector>
```
Comment (sokket) [inline src/Native/System.Native/pal_networking.cpp]: I'd prefer to try and keep the included alphabetical, unless there are dependencies between headers

**Comment by sokket** [comment]
Besides the minor NIT, and run clang-format on the native files that you changed, LGTM

File: src/Native/System.Native/pal_networking.cpp (hunk 1 above)
Comment (pgavlin) [inline src/Native/System.Native/pal_networking.cpp]: Ah, so that's how they were organized before. Noted.

**Comment by pgavlin** [comment]
Interop.opt{level,name}.cs have been deleted--since they were empty files, the diff looks a little strange.

**Pull request merged**